vmm/src/interrupt.rs

```diff
@@ -372,6 +373,19 @@ pub mod kvm {
                 kvm_route.u.msi.address_hi = cfg.high_addr;
                 kvm_route.u.msi.data = cfg.data;
 
+                #[cfg(target_arch = "aarch64")]
+                {
+                    // TODO:
+                    // Need to check Cap KVM_CAP_MSI_DEVID to determine whether devid is requred.
+                    // According to kernel doc Documentation/virtual/kvm/api.txt, "if this capability is not available,
+                    // userspace should never set the KVM_MSI_VALID_DEVID flag as the ioctl might fail".
+                    // However the Cap KVM_CAP_MSI_DEVID is not supported by kvm-ioctls yet.
+                    // Setting them unconditionally is a temporary solution, should be replaced once kvm-ioctls ready.
+                    // The code was tested on GICv3 which should be the most common case for modern hardware.
+                    kvm_route.flags = KVM_MSI_VALID_DEVID;
+                    kvm_route.u.msi.__bindgen_anon_1.devid = cfg.devid;
+                }
+
```

Comment:
I'd like to see the kvm-ioctls ch branch rebased before merging this PR.

Reply:
Hi, @rbradford , the dependency PR in rust-vmm/kvm-ioctls was merged, could you help update the fork again?

Reply:
@michael2012z Done.

Reply:
@rbradford , thank you. When I tried to implement the capability check, I realized that something is still missing. `KVM_CAP_MSI_DEVID` need to be checked per-VM, but in kvm-ioctls capacity check was only implemented for system (KVM), not for VM and VCPU. 

Sorry I didn't notice this when I submitted my last PR to rust-vmm. I will begin to add it tomorrow.

Do you prefer to use this existing workaround, or wait a while when the new API is settled?

Reply:
The PR was rebased to leverage the latest kvm-ioctls API. PTAL.